crates/ruma-common/src/events/encrypted.rs

```diff
@@ -23,11 +23,7 @@ pub struct EncryptedEventContent {
     pub encrypted: EncryptedContentBlock,
 
     /// Information about related events.
-    #[serde(
-        flatten,
-        skip_serializing_if = "Option::is_none",
-        deserialize_with = "super::room::encrypted::relation_serde::deserialize_relation"
-    )]
+    #[serde(rename = "m.relates_to", skip_serializing_if = "Option::is_none")]
     pub relates_to: Option<Relation>,
```

Comment:
This change is not strongly related to the main change of this PR, right? May be worth to make a PR for just this first? I.e. my understanding for why we can even get rid of `flatten` here is that `new_content` is part of the encrypted payload for encrypted events. That is, `flatten` never made a lot of sense here regardless of whether we keep data for unknown relations.